lite/provider_test.go

```diff
@@ -1,149 +1,140 @@
 	count := 10
 
-	// make a bunch of commits...
-	commits := make([]lite.FullCommit, count)
+	// Make a bunch of full commits.
+	fcz := make([]FullCommit, count)
 	for i := 0; i < count; i++ {
-		// two commits for each validator, to check how we handle dups
-		// (10, 0), (10, 1), (10, 1), (10, 2), (10, 2), ...
 		vals := keys.ToValidators(10, int64(count/2))
 		h := int64(20 + 10*i)
-		commits[i] = keys.GenFullCommit(chainID, h, nil, vals, appHash, []byte("params"), []byte("results"), 0, 5)
+		fcz[i] = keys.GenFullCommit(chainID, h, nil, vals, vals, appHash, []byte("params"), []byte("results"), 0, 5)
 	}
 
-	// check provider is empty
-	fc, err := p.GetByHeight(20)
-	require.NotNil(err)
-	assert.True(liteErr.IsCommitNotFoundErr(err))
-
-	fc, err = p.GetByHash(commits[3].ValidatorsHash())
+	// Check that provider is initially empty.
+	fc, err := p.LatestFullCommit(chainID, 1, 1<<63-1)
 	require.NotNil(err)
-	assert.True(liteErr.IsCommitNotFoundErr(err))
+	assert.True(lerr.IsErrCommitNotFound(err))
 
-	// now add them all to the provider
-	for _, s := range commits {
-		err = p.StoreCommit(s)
+	// Save all full commits to the provider.
+	for _, fc := range fcz {
+		err = p.SaveFullCommit(fc)
 		require.Nil(err)
-		// and make sure we can get it back
-		s2, err := p.GetByHash(s.ValidatorsHash())
-		assert.Nil(err)
-		assert.Equal(s, s2)
-		// by height as well
-		s2, err = p.GetByHeight(s.Height())
+		// Make sure we can get it back.
+		fc2, err := p.LatestFullCommit(chainID, fc.Height(), fc.Height())
 		assert.Nil(err)
-		assert.Equal(s, s2)
+		assert.Equal(fc.SignedHeader, fc2.SignedHeader)
+		assert.Equal(fc.Validators, fc2.Validators)
+		assert.Equal(fc.NextValidators, fc2.NextValidators)
 	}
 
-	// make sure we get the last hash if we overstep
-	fc, err = p.GetByHeight(5000)
+	// Make sure we get the last hash if we overstep.
+	fc, err = p.LatestFullCommit(chainID, 1, 5000)
 	if assert.Nil(err) {
-		assert.Equal(commits[count-1].Height(), fc.Height())
-		assert.Equal(commits[count-1], fc)
+		assert.Equal(fcz[count-1].Height(), fc.Height())
+		assert.Equal(fcz[count-1], fc)
 	}
 
-	// and middle ones as well
-	fc, err = p.GetByHeight(47)
+	// ... and middle ones as well.
+	fc, err = p.LatestFullCommit(chainID, 1, 47)
 	if assert.Nil(err) {
 		// we only step by 10, so 40 must be the one below this
 		assert.EqualValues(40, fc.Height())
 	}
 
 }
 
-// this will make a get height, and if it is good, set the data as well
-func checkGetHeight(t *testing.T, p lite.Provider, ask, expect int64) {
-	fc, err := p.GetByHeight(ask)
-	require.Nil(t, err, "GetByHeight")
+// This will make a get height, and if it is good, set the data as well.
+func checkLatestFullCommit(t *testing.T, p PersistentProvider, chainID string, ask, expect int64) {
+	fc, err := p.LatestFullCommit(chainID, 1, ask)
+	require.Nil(t, err)
 	if assert.Equal(t, expect, fc.Height()) {
-		err = p.StoreCommit(fc)
-		require.Nil(t, err, "StoreCommit")
+		err = p.SaveFullCommit(fc)
```

Comment:
Maybe it's better to save it explicitly when needed instead of doing it as a side effect.

Reply:
it's in the comment for that function, and would make the tests more clutterred... punting.